### ISSUE TITLE
OdbcConnection.ServerVersion may return null but docs says it will throw or return "" instead

### ISSUE THREAD

**Issue by krwq** [dotnet/runtime#44289]
See docs: https://docs.microsoft.com/en-us/dotnet/api/system.data.odbc.odbcconnection.serverversion?view=dotnet-plat-ext-3.1

but implementation OdbcConnectionOpen.ServerVersion will return null in some cases.

Either docs or implementation should be fixed here.

Found when applying nullable annotations

**Comment by roji** [comment]
Yeah, I ran across this during annotation: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnectionOpen.cs#L35 (based on https://github.com/dotnet/runtime/pull/689).

**Comment by krwq** [comment]
Yes, this is based on your comment, I'll be removing TODO-NULLABLEs or replacing them with issues or fixing in a PR soon
